src/decoder/mod.rs

```diff
@@ -129,17 +129,22 @@ pub struct Limits {
     /// 256MiB. If the entire image is decoded at once, then this will
     /// be the maximum size of the image. If it is decoded one strip at a
     /// time, this will be the maximum size of a strip.
-    decoding_buffer_size: usize,
+    pub decoding_buffer_size: usize,
     /// The maximum size of any ifd value in bytes, the default is
     /// 1MiB.
-    ifd_value_size: usize,
+    pub ifd_value_size: usize,
+    /// The purpose of this is to prevent all the fields of the struct from
+    /// being public, as this would make adding new fields a major version
+    /// bump.
+    _non_exhaustive: (),
```

Comment:
It is probably better to just use `#[doc(hidden)]`.

Reply:
Are there semantic difference to consider? The enum solutions use `doc(hidden)` since there is no such thing as a private variant so we can't really compare it directly. Is there prior art?

Reply:
This is the solution suggested in the relating rfc: https://rust-lang.github.io/rfcs/2008-non-exhaustive.html

Reply:
Of course, this is a `struct`, not an `enum`. :sweat_smile: